### ISSUE TITLE
PowerBi  A join operation cannot result in a table with duplicate column names ("owningbusinessunitname")

### ISSUE THREAD

**Issue by AravindSivanantham** [microsoft/coe-starter-kit#1291]
### Describe the issue

1. **Trying to import .pbit file we are facing the issues "Flow connection Reference Duplicate column name Owningbuisnessunitname"**
![Error1](https://user-images.githubusercontent.com/88816898/135485942-bec5734e-d607-4389-ae5b-52654cd910aa.jpg)

2.**We have upgrader the solution Center of Excellence - Core Components from  2.75 to 3.12** 
![Upgrade](https://user-images.githubusercontent.com/88816898/135487296-35d43328-7812-4594-8caa-5994f55e0eeb.JPG)
3. **After upgrade we import .pbit file the issue are not resolved**
App Connection Reference
A join operation cannot result in a table with duplicate column names ("owningbusinessunitname").
Flow Connection Reference
A join operation cannot result in a table with duplicate column names ("owningbusinessunitname").

Please help to resolve this issues **Flow connection Reference Duplicate column name Owningbuisnessunitname"**

Thanks

### Expected Behavior

To resolve the issues "Flow connection Reference Duplicate column name Owningbuisnessunitname"**

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

CenterofExcellenceCoreComponents_3_12_managed

### What app or flow are you having the issue with?

CoE PowerBI Dashboard OrgUrl Loading Issue

### Steps To Reproduce

1. **Trying to import .pbit file we are facing the issues "Flow connection Reference Duplicate column name Owningbuisnessunitname"**
![Error1](https://user-images.githubusercontent.com/88816898/135485942-bec5734e-d607-4389-ae5b-52654cd910aa.jpg)

**Comment by AravindSivanantham** [comment]
Thanks Manuelap for the reply, PowerBI Desktop version:Version: 2.95.983.0 64-bit (July 2021)
![PowerBi_version](https://user-images.githubusercontent.com/88816898/135498490-3b4c46cc-aa1d-4b2d-8cda-db76d9534c99.png)

**Comment by sanjuskc** [comment]
I am receiving the same error on latest version of Power BI desktop. Below are the details:

**Version: 2.97.921.0 64-bit (September, 2021)**

**Comment by manuelap-msft** [comment]
We're about to release the October version of the dashboard, could you check if you have the same issues with that version

- [CDSLegacy_CoEDashboard](https://github.com/microsoft/coe-starter-kit/blob/coe-october/CenterofExcellenceResources/Release/Collateral/CoEStarterKit/CDSLegacy_CoEDashboard_202110.pbit)
- [Dataverse_CoEDashboard](https://github.com/microsoft/coe-starter-kit/blob/coe-october/CenterofExcellenceResources/Release/Collateral/CoEStarterKit/Dataverse_CoEDashboard_202110.pbit)

For reference [Which dashboard to use](https://docs.microsoft.com/en-us/power-platform/guidance/coe/setup-powerbi#which-dashboard-to-use)

**Comment by sanjuskc** [comment]
Receiving same error with October release version of Dataverse_CoEDashboard. See screenshot below:
![image](https://user-images.githubusercontent.com/91690081/135578949-a0aa9e34-548c-4d83-afb7-bbb697a65f4a.png)

**Comment by FalkeSK** [comment]
Hey guys,

is there any solution regarding this?

I'm facing the same error message during our scheduled refresh.
I didn't changed anything so I'm wondering what could cause that issue?

BR

**Comment by danielstaleyedf** [comment]
I'm getting this too with:
- Power BI Version: 2.97.921.0 64-bit (September 2021)
- Core Components July 2021 (2.2.202)
- Dataverse_CoEDashboard_202107_v2

**Comment by manuelap-msft** [comment]
@danielstaleyedf you are on an old version of the Power BI Dashboard, and we're unable to support previous version. You can download the latest version here: https://github.com/microsoft/coe-starter-kit/releases/download/CoEStarterKit-October2021/Dataverse_CoEDashboard_202110.pbit

**Comment by manuelap-msft** [comment]
@FlkeSK what CoE Power BI dashboard version are you on (e.g. Dataverse_CoEDashboard_202109 or CDSLegacy_CoEDashboard_202109)?

**Comment by danielstaleyedf** [comment]
thanks @manuelap-msft. The same applies to the October 2021 version of the report :)

**Comment by manuelap-msft** [comment]
@danielstaleyedf Unfortunately we are unable to repo this error and have tried with multiple Power BI Desktop versions now. If you hit Close on the error, does data load?

**Comment by danielstaleyedf** [comment]
no it doesn't. I'm loading the report into this tenancy for the first time and Im using an older version of the core components (July 2021) so there might be some other issues.

I wonder if @AravindSivanantham previously had the report working and a BI report / Core Components subsequently caused the issue?

**Comment by manuelap-msft** [comment]
I can't say for sure if that's causing the issue, but typically Core Components and the dashboard have to be updated together as they're so closely linked.

**Comment by danielstaleyedf** [comment]
That's why I initially attempted to install the version that I did :)

**Comment by danielstaleyedf** [comment]
If it is helpful - I'm getting the same error on another tenant that we have where I previously successfully configured the Power BI report (July 2021 report and July 2021 core components) something must have changed in PowerBI Desktop or the Power Platform somewhere that stops this from working (certainly with older versions of the CoE core components).

**Comment by misskristine94** [comment]
Unfortunately, I'm getting the same error. Really embarrassing as I was setting up the CoE kit as part of a market engagement exercise with potential customers and ran into this issue while on the call. I temporarily deleted the 'Outer join' action in the right panel for both errors however this resulted in incorrect data being loaded.

Are there any updates please? I haven't had this issue before and I'm using the most recent version of Power BI desktop downloaded on Monday.

![image](https://user-images.githubusercontent.com/86930618/136065677-9a6f266e-b29a-423d-ae76-999d7c9e440f.png)

**Comment by misskristine94** [comment]
Thanks Emanuela, I appreciate it's not exactly an error :)

I've managed to fix it - when I went to both errors, rather than remove the 'Outer Join' action, I removed the actual column with 'owningbusinessunitname' instead for both App Connection Reference and Flow Connection Reference errors. The outer join action will still be performed and so the owningbusinessunitname column will be there and will not be duplicated. I've just been looking at the dashboard and all the data appears 100% correct.

Hopefully this will help someone 😊

**Comment by manuelap-msft** [comment]
That's amazing, thanks for finding this workaround! Impressive detective skills! 

I am very puzzled- when I open the report and load the data, I don't have a column called owningbusinessunitname (I just have owningbusinessunit which is already removed for App Connection Reference and Flow Connection Reference). 

Do you also have a field called owningbusinessunit and can you see that being removed? 
![image](https://user-images.githubusercontent.com/55705669/136076123-ab2cfddf-628e-4c6e-b5c3-b082ce9cc4c1.png)

I think for now this will definitely help everyone who has the same issue, and we can help with them with your instructions in getting the BI working. If you could share screenshots of where you made the changes and how the steps in Transform Data look, that would be super brilliant so we can point other users with this issue here! 

Thank you

**Comment by Nanditdhania** [comment]
Remove the column "owningbuisnessunitname" from the connector table and you're sorted.

**Comment by misskristine94** [comment]
> That's amazing, thanks for finding this workaround! Impressive detective skills!
> 
> I am very puzzled- when I open the report and load the data, I don't have a column called owningbusinessunitname (I just have owningbusinessunit which is already removed for App Connection Reference and Flow Connection Reference).
> 
> I think for now this will definitely help everyone who has the same issue, and we can help with them with your instructions in getting the BI working. If you could share screenshots of where you made the changes and how the steps in Transform Data look, that would be super brilliant so we can point other users with this issue here!
> 
> Thank you

Thank you Manuela! I guess all the police thrillers I've watched have taught me some detective skills! 😄 I am still baffled how that issue came up in the first place if I'm honest!

Apologies for the late reply but I've been out of the country and only just returned this morning. I can see that some users are still having the same issue as I did so I thought I'd clarify the steps to save everyone the time I spent figuring this out!

- When you get the error, please go to the App Connection Reference table.
- **Do not remove** any of the steps applied -removing any of those (see screenshot below) WILL break your report
![image](https://user-images.githubusercontent.com/86930618/137204478-c2fb9b04-a873-460e-9cd6-21b3b025f30c.png)
- within the table itself, find the column called 'owningbusinessunitname' and delete it 

![image](https://user-images.githubusercontent.com/86930618/137204993-c7a06e46-910b-4d67-b212-330555283ea2.png)

- do the same with the Flow Connection Reference table

And done! When I had a deeper dive, I noticed that one of the columns gets expanded and it already contains the a column with the same name which is why it was causing the error.

I really hope this helps someone!! 

Kristine 😊

**Comment by manuelap-msft** [comment]
I'm able to repo this in our tenant now, and this will be fixed for the December release. I may be able to upload the fixed .pbit file next week already as well.

**Comment by zwelding** [comment]
Thank you so much for the fix. If only my google searches pointed me here an hour ago!!

**Comment by manuelap-msft** [comment]
If you want to try out the new version before our release, the files are now available here: https://github.com/microsoft/coe-starter-kit/tree/coe-december/CenterofExcellenceResources/Release/Collateral/CoEStarterKit

I hope they work for you,
Manuela

**Comment by benstegink** [comment]
I was having the same issue with the 202110.pbit file. Just downloaded and tried the 202112.pbit file linked above by @manuelap-msft and that solved the issue for me.

**Comment by mikefactorial** [comment]
@AravindSivanantham This has been fixed in the latest release. Please install [the latest version of the toolkit](https://aka.ms/CoEStarterKitDownload) following the instructions for [installing updates](https://docs.microsoft.com/power-platform/guidance/coe/setup#installing-updates). Note that if you do not remove the unmanaged layers as described there you will not receive updates from us.

**Comment by bartsneary** [comment]
This worked perfectly for me today on the full Dataverse environment we have.  It failed with similar issues to the thread here on our Dataverse for Teams environment.  I had to manually remove owningbusinessunit and a few other columns it prompted me it was unhappy about since they didn't exist from the remove columns step of the Connector table in PBI.  After that, all worked well.

**Comment by Jenefer-Monroe** [comment]
Can you please post that issue for the Teams version of the dashboard?

**Comment by bartsneary** [comment]
It's the same dashboard - Dataverse_CoEDashboard_202112 - used both times.  It only had issues firing up when connected to my DV4T environment after updating it to v3.10 for the Core Components for Teams.  Happy to either log a new issue or put it someplace else if you tell me where exactly you'd like me to log it.

**Comment by Jenefer-Monroe** [comment]
Oh i had misunderstood. Yes pleas just post a new bug here, and say the above. We may need to investigate or ship sep dashboards.

**Comment by ajinkyaskulkarni** [comment]
Teams_CoEDashboard_Jan2022 has the same issue.

**Comment by Jenefer-Monroe** [comment]
please see #1724

**Comment by manuelap-msft** [comment]
Please see https://github.com/microsoft/coe-starter-kit/issues/1724#issuecomment-1009027307 for an new version that fixes this for Teams. We are still investigating this as currently it looks like this column (owningbusinessunitname) is only rolled out to Teams envs created after Jan 6 but not yet to Teams envs created before that. We need to further test this before updating the release, but the file provided in the above comment should unblock you.

**Comment by ajinkyaskulkarni** [comment]
@manuelap-msft, @Jenefer-Monroe  Thanks a lot for your quick response and new version with fix. I looked into the model and fixed the current report by adding remove column action for the column "owningbusinessunitname". I am also going to check the version with fix mentioned by you.
